clouds/aws/create_aws_network_resources.py

```diff
@@ -0,0 +1,793 @@
+        )
+        eip_list.append(eip["AllocationId"])
+    return eip_list
+
+
+def create_nat_gateway(
+    ec2_client, site_name, subnet_count, vpc_output, alloc_ids, subnet_type
+) -> list:
+    """
+    Creates NAT gateway to map private network to single public ip
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         subnet_count (int): subnet count to create nat gateway per subnet
+         site_name (str): vpc name to associate nat gateway
+
+    Returns:
+         list (nat_gateway_id_list): list containing nat gateway id
+    """
+    LOGGER.info("Creating Nat Gateway (1 per AZ) for each public subnet")
+
+    nat_gateway_id_list = []
+    for count in range(subnet_count):
+        nat_gateway = ec2_client.create_nat_gateway(
+            AllocationId=alloc_ids[count],
+            SubnetId=vpc_output["public_subnet_ids"][count],
+        )
+        ec2_assign_name_tag(
+            ec2_client=ec2_client,
+            res_id=nat_gateway["NatGateway"]["NatGatewayId"],
+            tag_name=f"{site_name}-nat-{subnet_type}{count+1}",
+        )
+        nat_gateway_id_list.append(nat_gateway["NatGateway"]["NatGatewayId"])
+    ec2_client.get_waiter("nat_gateway_available").wait(
+        NatGatewayIds=nat_gateway_id_list,
+    )
+    return nat_gateway_id_list
+
+
+def create_public_route_table(ec2_client, site_name, vpc_id) -> list:
+    """
+    Creates route table for public routes
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         site_name (str): vpc name
+         vpc_id (str): vpc Id of the associated vpc
+    Return:
+         str (rt_public_id): route table id
+    """
+    LOGGER.info("Creating  Public Route Tables")
+
+    rt_public = ec2_client.create_route_table(VpcId=vpc_id)
+    rt_public_id = rt_public["RouteTable"]["RouteTableId"]
+    rt_details = ec2_client.describe_route_tables(RouteTableIds=[rt_public_id])
+    try:
+        for sample in TimeoutSampler(
+            wait_timeout=300,
+            sleep=5,
+            func=is_resource_available,
+            aws_resource={"Resource": (rt_details["RouteTables"][0])["Routes"][0]},
+        ):
+            if sample is True:
+                break
+    except TimeoutExpiredError:
+        LOGGER.error(f" Route Table {rt_public_id} not available.")
+        raise
+
+    ec2_assign_name_tag(
+        ec2_client=ec2_client,
+        res_id=rt_public_id,
+        tag_name=f"{site_name}-rtb-public",
+    )
+    return rt_public_id
+
+
+def create_private_route_table(ec2_client, subnet_count, vpc_id, site_name) -> list:
+    """
+    Creates route table for private routes
+
+    Args:
+         ec2_client (botocore.client.EC2): aws botocore client
+         subent_count (int): no. of subnets
+         vpc_id (str): vpc Id of the associated vpc
+         site_name (str): vpc name
+
+    Return:
+         list: list containing subnet Ids
+    """
+    LOGGER.info("Creating  Private Route Tables")
```

Comment:
`Creating  Private` extra space, please remove it

Reply:
done